tasks/llvm.py

```diff
@@ -0,0 +1,85 @@
+from faasmtools.build import (
+    FAASM_BUILD_ENV_DICT,
+    FAASM_LOCAL_DIR,
+    WASM_LIB_INSTALL,
+)
+from faasmtools.env import (
+    LLVM_DIR,
+    LLVM_MAKEFILE,
+    LLVM_VERSION,
+    PROJ_ROOT,
+    USABLE_CPUS,
+    WASI_LIBC_DIR,
+)
+from invoke import task
+from os import environ
+from os.path import join
+from subprocess import run
+
+LLVM_BUILD_DIR = join(LLVM_DIR, "build")
+LLVM_BUILD_ENV_VARS = {
+    "FAASM_CPP_PROJ_ROOT": PROJ_ROOT,
+    "FAASM_LLVM_DIR": LLVM_DIR,
+    "FAASM_LLVM_BUILD_DIR": LLVM_BUILD_DIR,
+    "FAASM_LLVM_VERSION": LLVM_VERSION,
+    "FAASM_LOCAL_DIR_ENV": FAASM_LOCAL_DIR,
+    "FAASM_WASI_LIBC_DIR": WASI_LIBC_DIR,
+}
+
+
+def do_llvm_build(target, clean_target=None):
```

Comment:
This function is a common entrypoint to build the different targets in `LLVM.makefile` (most importantly, `all` or `libc`).

The main difference with how we did it before is that now, to re-build `libc` instead of going directly to `wasi-libc` and running `make` we run `make libc` from the `llvm-project` submodule. Under the hood, this does the same: `cd`s to `wasi-libc` and `make`, but it means that we now can rely on `LLVM.makefile` as the single source of truth.